src/gpl/src/nesterovPlace.cpp

```diff
@@ -707,10 +731,12 @@ NesterovPlace::doNesterovPlace(int start_iter) {
       }
     }
 
-    // minimum iteration is 50
-    if( iter > 50 && sumOverflow_ <= npVars_.targetOverflow) {
-      log_->report("[NesterovSolve] Finished with Overflow: {:.6f}", sumOverflow_);
-      break;
+    // 1. to non-trivial design --> assign minimum 50 iterations.
+    // 2. to trivial design --> not assign minimum iterations.
+    if( pb_->insts().size() >= 500 && iter >= 50 && sumOverflow_ <= npVars_.targetOverflow ||
+        pb_->insts().size() < 500 && sumOverflow_ <= npVars_.targetOverflow ) {
+       log_->report("[NesterovSolve] Finished with Overflow: {:.6f}", sumOverflow_);
+       break;
```

Comment:
See other comment about magic numbers.

Reply:
Why do we need a 50 iteration limit in the first place?

Reply:
There are placement gaps between the result of BiCGSTAB QP solver and the initial Nesterov iteration (when wirelength force is very high and density force is very low). The insts locations are abruptly changed, so there is a possibility to reach the < 0.1 overflows in a coincidence. The iteration < 50 blocks such situations.

Reply:
What does it matter if it is a coincidence that you converge early?  What does keeping improve?

Reply:
No. in this case, RePlAce doesn't converge at all.  It is tool noise for non-trivial designs, so it just stopped with the noised random placements and the overflow reached <0.1 in a coincidence.  We know that RePlAce didn't converge < iter 50 for non-trivial designs, like aes. (minimum #iter >= 200)

Reply:
I just tried aes on ng45 & sky130hd without this limit and neither converges early.  It is hard for me to imagine how on a non-trivial design you could get a jump that produced convergence.  On a trivial design I'm not sure it matters.

Reply:
Can we keep this setup in this PR and make a separate PR to remove this iter >= 50 flag to test in deeply?  I remember there was a corner testcase that this happened, but I couldn't find it right now...

Reply:
I've erased iter >= 50 flags.